v-next/hardhat/src/internal/builtin-plugins/solidity-test/task-action.ts

```diff
@@ -0,0 +1,71 @@
+import type { NewTaskActionFunction } from "../../../types/tasks.js";
+
+import { spec } from "node:test/reporters";
+
+import { buildSolidityTestsInput, runAllSolidityTests } from "./helpers.js";
+
+const runSolidityTests: NewTaskActionFunction = async (_arguments, hre) => {
+  await hre.tasks.getTask("compile").run({ quiet: false });
+
+  console.log("\nRunning Solidity tests...\n");
+
+  const specReporter = new spec();
```

Comment:
The use of Node test runners spec reporter is similarly taken directly from the v2 plugin example.
We should feel free to build on it or replace as we see fit.